## **Model Inference**
---
Pada tahap ini, saya akan menggunakan model machine learning yang telah dilatih (`best_los_gbt_model.pkl`) untuk memprediksi rata-rata lama tinggal (`length_of_stay_avg`) pasien baru di rumah sakit.

In [12]:
# Import Libraries
import pickle
import pandas as pd

In [13]:
# Load model
with open('best_los_gbt_model.pkl', 'rb') as file_1:
  model = pickle.load(file_1)

In [14]:
# Menyiapkan Data Inferensi Pasien Baru
pasien_baru = {
    # Kolom yang akan digunakan oleh pipeline
    'admission_count': 2,
    'condition_type': 'Asthma',
    'patient_age_group': '18-45',
    'patient_gender': 'Male',
    'readmission_count': 1.0,
    'severity_level': 'Severe',
    'comorbid_conditions_count': 1.0,
    'daily_medication_dosage': 15.0,
    'emergency_visit_count': 0.0,
    'admission_month': 'January',
    'admission_day_of_week': 'Friday',
    'primary_diagnosis_code': 'Other',

    # Kolom yang akan didrop sebelum melatih pipeline
    'hospital_name': 'Mecca City Hospital',
    'seasonal_indicator': 'Winter',
    'admission_year': 2025}

df_inferensi = pd.DataFrame(pasien_baru, index=[0])
df_inferensi.head()


admission_count condition_type patient_age_group patient_gender  \
0                2         Asthma             18-45           Male   

   readmission_count severity_level  comorbid_conditions_count  \
0                1.0         Severe                        1.0   

   daily_medication_dosage  emergency_visit_count admission_month  \
0                     15.0                    0.0         January   

  admission_day_of_week primary_diagnosis_code        hospital_name  \
0                Friday                  Other  Mecca City Hospital   

  seasonal_indicator  admission_year  
0             Winter            2025

In [15]:
# Daftar kolom yang akan dihapus
columns_to_drop = [
    'hospital_name', 
    'seasonal_indicator',
    'admission_year'       
]
# Menghapus kolom dari X_train_capped
df_inferensi.drop(columns=columns_to_drop)

admission_count condition_type patient_age_group patient_gender  \
0                2         Asthma             18-45           Male   

   readmission_count severity_level  comorbid_conditions_count  \
0                1.0         Severe                        1.0   

   daily_medication_dosage  emergency_visit_count admission_month  \
0                     15.0                    0.0         January   

  admission_day_of_week primary_diagnosis_code  
0                Friday                  Other

In [16]:
# Melakukan Prediksi pada Data Baru 
prediksi_los = model.predict(df_inferensi)
        
df_inferensi['predicted_length_of_stay_avg'] = prediksi_los
print("Hasil Prediksi Length of Stay (rata-rata hari):", prediksi_los[0])
print("\nData inferensi dengan hasil prediksi:")
df_inferensi

Hasil Prediksi Length of Stay (rata-rata hari): 4.9925204371802625

Data inferensi dengan hasil prediksi:


admission_count condition_type patient_age_group patient_gender  \
0                2         Asthma             18-45           Male   

   readmission_count severity_level  comorbid_conditions_count  \
0                1.0         Severe                        1.0   

   daily_medication_dosage  emergency_visit_count admission_month  \
0                     15.0                    0.0         January   

  admission_day_of_week primary_diagnosis_code        hospital_name  \
0                Friday                  Other  Mecca City Hospital   

  seasonal_indicator  admission_year  predicted_length_of_stay_avg  
0             Winter            2025                       4.99252

Pasien baru ini memiliki karakteristik: `admission_count` = 2, `condition_type`= Asthma, `patient_age_group`= 18-45, `patient_gender`= Male, `readmission_count`= 1.0, `severity_level`= Severe, `comorbid_conditions_count`= 1.0, `daily_medication_dosage`= 15.0, dan `emergency_visit_count`= 0.0, serta detail waktu lainnya.

Model ini **memprediksi bahwa rata-rata lama tinggal (`predicted_length_of_stay_avg`) untuk pasien dengan profil ini adalah sekitar 4.99 hari.**

Interpretasi hasil prediksi ini harus dilakukan dengan sangat hati-hati. Mengingat MAE model sekitar 1.19 hari, prediksi 4.99 hari tersebut memiliki rentang ketidakpastian yang signifikan (misalnya, aktualnya bisa jadi antara 3.8 hingga 6.18 hari). Dan dengan R2 negatif, keyakinan terhadap hasil prediksi ini sangat rendah. Model belum bisa dipercaya untuk membedakan secara tepat pasien yang akan tinggal 4 hari versus 5 hari atau 6 hari.